ui/src/app/components/apps-panel/utils.tsx

```diff
@@ -108,7 +108,7 @@ export const toAppType = (type: UIAppType): AppType =>
   );
 
 export const findApp = (
-  apps: UserAppEnvironment[],
+  apps: UserAppEnvironment[] | null | undefined,
```

Comment:
Thinking out loud, I wonder if we could create a type alias for null or undefined when they have particular meanings, like:
```
type NotYetLoaded = null
```

Reply:
I like this idea, I'll consider it for future typings. First we should find ways to reduce the need for `null` and `undefined` in the first place (like making our stores strongly-typed).